Moving #defines to proper file

llersch opened the pull request:
Defining variable in the source files might generate unpredictable bugs if
dependency between compilation units change.

For example, #define BP_ALTERNATE_CB_LATCH was done in bf_tree.h, but used
also in bf_tree_cb.h. If someone else included bf_tree_cb.h without #define,
weird behaviors would happen.

llersch committed:
Moving #defines to proper file

Defining variable in the source files might generate unpredictable bugs if
dependency between compilation units change.

For example, #define BP_ALTERNATE_CB_LATCH was done in bf_tree.h, but used
also in bf_tree_cb.h. If someone else included bf_tree_cb.h without #define,
weird behaviors would happen.

llersch committed:
Cleaner should acquire latch before modifying cb

caetanosauer commented:
Do you think you could put these flags all in the cmake configuration and delete the shore.def file? I think that would be a more permanent solution.

llersch committed:
Removing #define BP_ALTERNATE_CB_LATCH

Alternate control blocks and latches in the bufferpool is now the one and only
behavior supported by Zero. We are not supporting older layout anymore.

llersch committed:
Removing obsolete code in bf_tree.h

The code removed was not being used or referenced anywhere else in the code.

llersch committed:
Removing #define USE_TLS_ALLOCATOR

This was being defined by default anyway. No documentation, be the alternative
to the TLC allocator was called "naive allocator", so it is probably safe to
discard it.

llersch committed:
Removing #define KEEP_LOG_PARTITIONS

It was not being used anywhere else

llersch committed:
Removing #undef USE_SSMTEST

It was not being used anywhere else in the code
